storage/database/leveldb_database.go

```diff
@@ -92,12 +92,19 @@ type levelDB struct {
 	fn string      // filename for reporting
 	db *leveldb.DB // LevelDB instance
 
-	compTimeMeter   metrics.Meter // Meter for measuring the total time spent in database compaction
-	compReadMeter   metrics.Meter // Meter for measuring the data read during compaction
-	compWriteMeter  metrics.Meter // Meter for measuring the data written during compaction
-	diskReadMeter   metrics.Meter // Meter for measuring the effective amount of data read
-	diskWriteMeter  metrics.Meter // Meter for measuring the effective amount of data written
-	blockCacheGauge metrics.Gauge // Gauge for measuring the current size of block cache
+	writeDelayCountMeter    metrics.Meter // Meter for measuring the cumulative number of write delays
+	writeDelayDurationMeter metrics.Meter // Meter for measuring the cumulative duration of write delays
```

Comment:
Why are these `cumulative` number and duration? The meters measures the current values.
```
		db.writeDelayCountMeter.Mark(int64(s.WriteDelayCount - prevWriteDelayCount))
		db.writeDelayDurationMeter.Mark(int64(s.WriteDelayDuration - prevWriteDelayDuration))
```

Reply:
> 1. What is `WritePaused` in `leveldb.DBStats`? Does levelDB pause sometimes? How about printing a log when the DB stops?

When the compaction is in-progress, write operation is paused. Please check below codes.

```
func (db *DB) flush(n int) (mdb *memDB, mdbFree int, err error) {
...
		case tLen >= pauseTrigger:
			delayed = true
			// Set the write paused flag explicitly.
			atomic.StoreInt32(&db.inWritePaused, 1)
			err = db.compTriggerWait(db.tcompCmdC)
			// Unset the write paused flag.
			atomic.StoreInt32(&db.inWritePaused, 0)
			if err != nil {
				return false
			}
...
```
```
// This will trigger auto compaction and/or wait for all compaction to be done.
func (db *DB) compTriggerWait(compC chan<- cCmd) (err error) {
	ch := make(chan error)
	defer close(ch)
	// Send cmd.
	select {
	case compC <- cAuto{ch}:
	case err = <-db.compErrC:
		return
	case <-db.closeC:
		return ErrClosed
	}
	// Wait cmd.
	select {
	case err = <-ch:
	case err = <-db.compErrC:
	case <-db.closeC:
		return ErrClosed
	}
	return err
}
```

> 2. What are `LevelSizes`, `LevelTablesCounts`, `LevelRead`, `LevelWrite`, `LevelDurations` in `leveldb.DBStats`? And how are they different from other stats?

As you might have heard of, LevelDB consists of different levels of data. So, those stats with "Level" suffix provide statistics from each level. Below are some translated LevelDB documents.

https://github.com/ground-x/documents/blob/master/klaytn/databases/leveldb-doc-impl.md
https://github.com/ground-x/documents/blob/master/klaytn/databases/leveldb-doc-index.md

Reply:
Thanks for the kind comment 👍